### DISCUSSION TITLE
AWS: Load tags as an EC2Instance label

### DISCUSSION THREAD

**Discussion by pavelb97** [cartography-cncf/cartography#621] [Q&A] [answered]
Hi all,
I'm using Cartography to map my AWS infrastructure. Loving the tool as it is pretty much ready to go out of the box!

My only issue is that tags do not show as part of an EC2 node labels on neo4j. These AWS tags are extremely helpful for identifying the resource in human readable format and I would like to use them as labels rather than nodes.

Can anyone point me in the direction of how I could load tags or even a specific tag key directly as a property attribute?

Just as a reference I've added a snipped of the property values displayed by an EC2Instance resource below in case I'm not being clear:
![image](https://user-images.githubusercontent.com/48152630/118135464-fba4a780-b3fa-11eb-97b6-fd48b1762f6d.png)

Any advice is much appreciated!

**Comment by ramonpetgrave64** [comment]
Hey. I just saw this. We usually have github issues, not Discussion threads. But are you referring to AWSTags? we have a sync module specifically for importing and attaching nodes with data fetched from AWS's resource tagging api. Let's continue the discussion in a github issue you create.

**Comment by achantavy** [comment, marked as answer]
I'll answer this question (even though it's like 2 years after, oops sorry):

You can definitely query for tags attached to an EC2 instance like this:

```cypher
match (i:EC2Instance)--(t:AWSTag) return i.instanceid, t.key, t.value limit 30;
```

See the schema docs for more details on the nodes and rels that we define. The sync job that populates this is called `resourcegroupstaggingapi`.
